### PR TITLE
Run tests with travis

### PR DESCRIPTION
**Pull request by lmignon**
Since travis need to be enabled for this repository (https://travis-ci.org/profile/anybox) you can see the result here https://travis-ci.org/acsone/anybox.recipe.odoo
For coverage, the project must be registered on https://coveralls.io/

### CODE REVIEW

**Comment by petrus-v** [comment]
@lmignon,

Thanks for setting!

I enable travis for this project, I gonna look about coverall!
cc @gracinet , @ccomb

File: .travis.yml [hunk 1]
```diff
@@ -0,0 +1,27 @@
...
+  - "2.7"
+
+virtualenv:
+  system_site_packages: true
+
+install:
+  - pip install .
+  - pip install coveralls
+  - pip install flake8
+  - pip install bzr
```
Comment (petrus-v) [inline .travis.yml]: I'm not an expert about travis configuration, I guess we need add some requirements according documentation or the doc is obsolete?

...
Install nose, flake8 and, optionally, coverage:
```
recipe-env/bin/pip install nose coverage flake8==2.0 \
                   pep8==1.4.6 mccabe==0.2.1 pyflakes==0.7.3
```
...

Maybe we could user a dev requirement file ? Am I wrong?

**Comment by petrus-v** [comment]
coveralls is on a read only mode while work on updating system, I'll do it later.

File: .travis.yml (hunk 1 above)
Comment (lmignon) [inline .travis.yml]: doc seems to be obsolete. We can also put all these requirements in a requirements.txt file and call pip install -f requirements.txt

Comment (petrus-v) [inline .travis.yml]: :+1: to use requirement file as it let developers and any other tools to get the same env using the same file avoiding maintaining documentation / config files and so on.

Can you change documentation in same PR?

**Comment by lmignon** [comment]
@petrus-v Travis is green! You are now able to install all the required dependencies to dev, tests and run quality tools (flake8, coverage) with one command:
```shell
virtualenv recipe-env
git clone https://github.com/anybox/anybox.recipe.odoo
recipe-env/bin/pip install -e .[dev]
```

**Comment by gracinet** [comment]
@lmignon this is great :-) I wonder though about `system-site-packages=true`, isn't that a potential source of side effects ? Maybe I'm being too warry in a context with container insulation ?

**Comment by gracinet** [comment]
nitpick: I think it'll be useful to dump the versions of all installed packages in the build output.
Thanks again !

**Comment by lmignon** [comment]
@gracinet Thank you for the review. I've addressed all your comments:
* Travais no more use the system site packages
* Version of installed packages is dumped in the build output  https://travis-ci.org/anybox/anybox.recipe.odoo/builds/72787385#L3891

**Comment by petrus-v** [comment]
Hi,

Thanks again @lmignon.

I've enable coveralls on that repo.

**Pull request merged**